spring-webflux/src/main/java/org/springframework/web/reactive/config/ResourceChainRegistration.java

```diff
@@ -81,15 +81,13 @@ public ResourceChainRegistration(boolean cacheResources, @Nullable Cache cache)
 	public ResourceChainRegistration addResolver(ResourceResolver resolver) {
 		Assert.notNull(resolver, "The provided ResourceResolver should not be null");
 		this.resolvers.add(resolver);
-		if (resolver instanceof VersionResourceResolver) {
-			this.hasVersionResolver = true;
-		}
-		else if (resolver instanceof PathResourceResolver) {
-			this.hasPathResolver = true;
-		}
-		else if (resolver instanceof WebJarsResourceResolver) {
-			this.hasWebjarsResolver = true;
-		}
+
+        switch (resolver) {
+            case VersionResourceResolver ignored -> this.hasVersionResolver = true;
+            case PathResourceResolver ignored -> this.hasPathResolver = true;
+            case WebJarsResourceResolver ignored -> this.hasWebjarsResolver = true;
+        }
+		
```

Comment:
These changes break the build.

Please revert these changes.

[Pattern Matching for switch](https://openjdk.org/jeps/441) did not become an "enabled" feature until Java 21, and we have a Java 17 baseline.

More importantly, please ensure that you actually run the build before submitting a PR.